<?xml version="1.0"?>
<launch>
    <!-- config file path  -->
    <arg name="rviz_settings" default="$(find pedestrian_state_predictor)/config/rviz/debug_with_prius.rviz"/>
    <arg name="pedestrian_state_predictor_settings" default="$(find pedestrian_state_predictor)/config/param/pedestrian_state_predictor_debug.yaml"/>

    <!-- node launch -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_settings)"/> -->
    <node pkg="pedestrian_state_predictor" type="pedestrian_state_predictor_node" name="pedestrian_state_predictor" output="screen">
        <rosparam command="load" file="$(arg pedestrian_state_predictor_settings)"/>
        <remap from="/robot_odom" to="/pedsim_simulator/robot_position"/>
    </node>
    
    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="planner_rviz" args="-d $(arg rviz_settings)"/>

    <!-- =============== Simulator =============== -->
    <!-- param -->
    <arg name="kbd_teleop" default="false"/>
    <arg name="rqt_teleop" default="false"/>
    <arg name="visualize" default="true"/>
    <arg name="with_robot" default="true"/>

    <arg name="simulation_factor" default="1.0"/> <!-- Speed up -->
    <arg name="update_rate" default="25.0"/> <!-- Hz -->
    <arg name="ns" default=""/>
    <arg name="robot_description" default="$(find experimental_package)/models/zoe/urdf/$(arg ns)/zoe.urdf"/>

    <arg name="pose_initial_x" default="0.0"/>
    <arg name="pose_initial_y" default="-20.0"/>
    <arg name="pose_initial_theta" default="1.57"/>
    <arg name="max_robot_speed" default="2.5"/>

    <!-- launch -->
    <include file="$(find pedsim_simulator)/launch/simulator.launch">
        <arg name="kbd_teleop" value="$(arg kbd_teleop)"/>
        <arg name="rqt_teleop" value="$(arg rqt_teleop)"/>
        <arg name="scene_file" value="$(find experimental_package)scenarios/empty_area/empty_area.xml"/>
        <arg name="with_robot" value="$(arg with_robot)"/>
        <arg name="robot_description" value="$(arg robot_description)"/>
        <arg name="simulation_factor" value="$(arg simulation_factor)"/>
        <arg name="update_rate" value="$(arg update_rate)"/>
        <arg name="default_queue_size" value="10"/>
        <arg name="max_robot_speed" value="$(arg max_robot_speed)"/>
        <arg name="robot_mode" value="2"/>
        <arg name="enable_groups" value="true"/>
        <arg name="group_size_lambda" default="0.8"/>
        <arg name="groups_couples_proportion" default="0.05"/>
        <arg name="groups_friends_proportion" default="0.1"/>
        <arg name="groups_families_proportion" default="0.05"/>
        <arg name="groups_coworkers_proportion" default="0.8"/>
        <arg name="pose_initial_x" value="$(arg pose_initial_x)"/>
        <arg name="pose_initial_y" value="$(arg pose_initial_y)"/>
        <arg name="pose_initial_theta" value="$(arg pose_initial_theta)"/>
        <arg name="ns" value="$(arg ns)"/>
  </include>

    <!-- Create a transform sender for linking these frames: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id period_in_ms -->
    <node name="tf_footprint_base" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 $(arg ns)/base_footprint $(arg ns)/base_link 100"/>

    <!-- Visualizer -->
    <include file="$(find pedsim_visualizer)/launch/visualizer.launch">
        <arg name="ns" value="$(arg ns)"/>
    </include>

    <!-- Rviz -->
    <node pkg="rviz" type="rviz" name="sim_rviz" args="-d $(find pedsim_simulator)/rviz/with_robot$(arg ns).rviz" if="$(arg visualize)"/>
</launch>